projects/observability/src/shared/dashboard/data/graphql/calls-percentage/calls-percentage-data-source.model.ts

```diff
@@ -0,0 +1,83 @@
+import { forkJoinSafeEmpty, getPercentage } from '@hypertrace/common';
+import {
+  GraphQlDataSourceModel,
+  MetricAggregation,
+  MetricAggregationType,
+  MetricHealth
+} from '@hypertrace/distributed-tracing';
+import { Model, ModelProperty, STRING_PROPERTY } from '@hypertrace/hyperdash';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ExploreSpecification } from '../../../../graphql/model/schema/specifications/explore-specification';
+import { ExploreSpecificationBuilder } from '../../../../graphql/request/builders/specification/explore/explore-specification-builder';
+import {
+  ExploreGraphQlQueryHandlerService,
+  EXPLORE_GQL_REQUEST,
+  GraphQlExploreResponse
+} from '../../../../graphql/request/handlers/explore/explore-graphql-query-handler.service';
+
+@Model({
+  type: 'calls-percentage-data-source'
+})
+export class CallsPercentageDataSourceModel extends GraphQlDataSourceModel<MetricAggregation> {
+  @ModelProperty({
+    key: 'context',
+    type: STRING_PROPERTY.type,
+    required: true
+  })
+  public context!: string;
+
+  @ModelProperty({
```

Comment:
Same comment as the earlier PR - why do we need specific data sources for each use case - they're all the same, taking in two metrics and dividing them.

Reply:
difference is that this data source needs filters to be applied on the numerator, where as the error-percentage calculation do not.
I tried to keep this generic enough within the context of calls. This data source can be reused for various cases in the numerator like you see in the home dashboard in this PR.

Reply:
I did think of defining a generic percentage data source model, but there are too many possibilities around that to cover. Some percentage calculations make calls in the same context for both numerator and denominator, some others don't. Some need filters on the numerator, some would need on denominator, and some might need at both places.

Reply:
@aaron-steinfeld if there are no other comments apart from the specificity of data sources, shall I merge this and take up that refactor after?

Reply:
sure